llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -704,84 +838,53 @@ int main(int argc, char **argv) {
   if (OutputFilename.getNumOccurrences() == 0)
     OutputFilename = (Twine(sys::path::stem(InputFilename)) + ".files").str();
 
-  std::map<StringRef, std::vector<Function *>> GlobalsSet;
-
-  if (DoSplit || DoSymGen) {
-    KernelMapEntryScope Scope = Scope_Global;
-    if (DoSplit) {
-      if (SplitMode == SPLIT_AUTO)
-        Scope = selectDeviceCodeSplitScopeAutomatically(*MPtr);
-      else
-        Scope =
-            SplitMode == SPLIT_PER_KERNEL ? Scope_PerKernel : Scope_PerModule;
-    }
-    collectKernelModuleMap(*MPtr, GlobalsSet, Scope);
-  }
+  std::unique_ptr<Module> SyclModule;
+  std::unique_ptr<Module> EsimdModule;
+  std::tie(SyclModule, EsimdModule) = splitSyclEsimd(std::move(M));
+
+  // Do we have both Sycl and Esimd kernels?
+  bool SyclAndEsimdKernels = SyclModule && EsimdModule;
+
+  TableFiles SyclTblFiles =
+      processOneModule(std::move(SyclModule), false, SyclAndEsimdKernels);
+
+  // TODO: Currently -ir-output-only option doesn't work when there is a mix
+  // of SYCL and ESIMD kernels. The option requires a single LLVM IR output
+  // file, which we cannot provide since SYCL and ESIMD kernels should always
+  // be split.
+  // Today, -ir-output-only is a FPGA use case, while ESIMD kernels only appear
+  // in programs that run on a GPU, so they are orthogonal. But once there
+  // would be a use case for -ir-output-only on a module that has SYCL and ESIMD
+  // code, we need to address it.
+  if (IROutputOnly)
```

Comment:
it seems worthwhile to issue a warning to let user know that ESIMD part if ignored.

Reply:
I introduced a new `-split-esimd` option, which is not compatible with `-ir-output-only`.